modules/tensor_mechanics/src/materials/ComputeInstantaneousThermalExpansionFunctionEigenstrain.C

```diff
@@ -53,36 +53,32 @@ ComputeInstantaneousThermalExpansionFunctionEigenstrainTempl<is_ad>::initQpState
 }
 
 template <bool is_ad>
-void
-ComputeInstantaneousThermalExpansionFunctionEigenstrainTempl<is_ad>::computeThermalStrain(
-    GenericReal<is_ad> & thermal_strain, Real * dthermal_strain_dT)
+ValueAndDerivative<is_ad>
+ComputeInstantaneousThermalExpansionFunctionEigenstrainTempl<is_ad>::computeThermalStrain()
 {
   if (this->_t_step > 1)
     _step_one = false;
 
-  const auto & current_temp = this->_temperature[_qp];
-
-  const Real & old_thermal_strain = _thermal_strain_old[_qp];
-
+  const auto & current_temp = this->_temperature_prop[_qp];
   const auto & old_temp =
-      (_step_one ? this->_stress_free_temperature[_qp] : this->_temperature_old[_qp]);
+      (_step_one ? this->_stress_free_temperature[_qp] : this->_temperature_old_prop[_qp]);
   const auto delta_T = current_temp - old_temp;
 
+  const Real & old_thermal_strain = _thermal_strain_old[_qp];
+
   const auto alpha_current_temp = _thermal_expansion_function.value(current_temp);
   const auto alpha_old_temp = _thermal_expansion_function.value(old_temp);
 
-  thermal_strain = old_thermal_strain + delta_T * 0.5 * (alpha_current_temp + alpha_old_temp);
+  auto thermal_strain = old_thermal_strain + delta_T * 0.5 * (alpha_current_temp + alpha_old_temp);
 
   // store this for use in the next timestep (no derivatives needed)
   _thermal_strain[_qp] = MetaPhysicL::raw_value(thermal_strain);
 
-  if constexpr (!is_ad)
-  {
-    mooseAssert(dthermal_strain_dT, "Internal error. dthermal_strain_dT should not be nullptr.");
-    *dthermal_strain_dT = alpha_current_temp;
-  }
+  if constexpr (is_ad)
+    return thermal_strain;
   else
-    libmesh_ignore(dthermal_strain_dT);
+    // return {thermal_strain, 0.5 * (alpha_current_temp + alpha_old_temp)};
+    return {thermal_strain, alpha_current_temp};
```

Comment:
@bwspenc , this derivative seem wrong to me. The derivative should be what it is in the commented out version, but that leads to two tests diffing.